server/src/main/java/org/opensearch/cluster/routing/allocation/allocator/BalancedShardsAllocator.java

```diff
@@ -205,10 +263,18 @@ private void setPreferPrimaryShardBalance(boolean preferPrimaryShardBalance) {
         this.weightFunction.updateRebalanceConstraint(INDEX_PRIMARY_SHARD_BALANCE_CONSTRAINT_ID, preferPrimaryShardBalance);
     }
 
+    private void setPreferPrimaryShardRebalance(boolean preferPrimaryShardRebalance) {
```

Comment:
lets store the preferPrimaryShardRebalance in a variable as well and create a method that takes no argument, but uses the variable which is called after each update.